crates/server/src/store/file/authority.rs

```diff
@@ -288,5 +344,32 @@ mod tests {
             RData::A(ip) => assert_eq!(Ipv4Addr::new(127, 0, 0, 1), *ip),
             _ => panic!("wrong rdata type returned"),
         }
+
+        /// some options how to make it happen:
+        /// 1. parser:parse_file that builds lexer's as it needs
+        ///    (mut lexer should be replaced with current_lexer and
+        ///    stack of "previous" lexer's, swap when we finished with
+        ///    the lexer)
+        /// 2. pub function to "prepare" file before building lexer,
+        ///    going line by line and simply injecting content of a different
+        ///    file before even creating lexer
+        let include_lookup = block_on(Authority::lookup(
+            &authority,
+            &LowerName::from_str("include.alias.example.com.").unwrap(),
+            RecordType::A,
+            false,
+            SupportedAlgorithms::new(),
+        ))
+        .expect("INCLUDE lookup failed");
+
+        match include_lookup
+            .into_iter()
+            .next()
+            .expect("A record not found in authity")
+            .rdata()
+        {
+            RData::A(ip) => assert_eq!(Ipv4Addr::new(127, 0, 0, 5), *ip),
```

Comment:
I'm assuming this is POC test?

There a bunch of different options around testing the FileAuthority, let me know if you want pointers.

Reply:
Actually, I just copy-pasted the test that was there (to check that new IP address from INCLUDE file could be looked up successfully). Sure, if there are better examples how to do this, - would love the reference ;)

Reply:
no, this is fine.